src/NuGet.Core/NuGet.Protocol/Events/ProtocolDiagnostics.cs

```diff
@@ -17,6 +17,14 @@ public static class ProtocolDiagnostics
 
         public static event ProtocolDiagnosticsNupkgCopiedEventHandler NupkgCopiedEvent;
 
+        public delegate void HttpCacheHitEventHandler();
+
+        public static event HttpCacheHitEventHandler HttpCacheHitEvent;
+
+        public delegate void HttpCacheMissEventHandler(bool cacheByPass, bool isExpired);
+
+        public static event HttpCacheMissEventHandler HttpCacheMissEvent;
```

Comment:
This doesn't feel in line with the other ProtocolDiagnostic events. Both hit and miss are cache events, so I think it makes sense to have a `ProtocolDiagnosticHttpCacheEvent`, and then a single new event in this class.